.gitignore

```diff
@@ -17,3 +17,8 @@ static/stylesheets/no-mq.css
 static/stylesheets/style.css
 __pycache__
 *.db
+
+
+# Ignore Microsoft Visual Studio Code's and Jetbrain's Pycharm Files
+.vscode/
+.idea/
```

Comment:
This seems kind of unnecessary, as the repository contains no `.vscode` or `.idea` directories, nor does anything in the repo generate such files.

Presumably those are added because you use tools that generate them locally. But if the project's `.gitignore` is going to contain exclusions for your personal metadata files from your personal choice of editor(s), then isn't it going to end up having to contain exclusions for **every** possible tool metadata file? No matter what it is, no matter how obscure, guarantee someone somewhere is generating it in their clone right now.

Fortunately, Git has a built-in solution for this. If you want to protect against accidentally adding files/directories that you generate locally,  when creating  your **local** commits, you should instead add those exclusions to the file `.git/info/exclude`. That's where local additions to the ignore list belong.

The Git developers have unfortunately not been very successful at disseminating this information, though it is presented right at the top of the [`gitignore{5}`](https://man7.org/linux/man-pages/man5/gitignore.5.html) man page.

(The problem is, `gitignore(5)` doesn't explain **why** some things belong in `.gitignore` and other things belong in `.git\info\exclude`. `.gitignore` gets a whole paragraph of explanation, and then it's followed by  a bullet point: "Patterns read from `$GIT_DIR/info/exclude`." That's it, end of item.)

So, what I'm saying is: Not your fault, extremely common mistake. But, `.gitignore` is still the wrong place for such additions. It's meant for (quoting from `gitignore(5)`) "patterns for files generated as part of the project build.".

Reply:
Good day, 

Thanks for getting back to me on this PR. I have updated ..gitignore file and I have made the changes as you've instructed. Please do reach out if you have any questions or clarifications.